NodeJS/src/AgentClient.ts

```diff
@@ -0,0 +1,413 @@
+        return this.session.post(
+            'spokenAbc/extensions/next/' + subAccountId + '/type/' + type)
+            .then((response: { data: any }) => {
+                return response.data
+            })
+    }
+
+    async getSkillIds(subAccountId: string) {
+        return this.session.get(
+            'spokenAbc/skills/multiClientSkills?clientIds=' + subAccountId + '&skillType=AGENT')
+            .then((response: { data: { [x: string]: { [x: string]: any } } }) => {
+                // console.log(response)
+                let skillResponses = response.data['skillResponses'][subAccountId];
+                return skillResponses.map((skillResponse: { id: any }) => skillResponse.id);
+            })
+    }
+
+    async getSkillNumbers() {
+        await this.session.login();
+
+        let subAccountId = await this.session.getSubAccount()
+            .then((response: { id: any }) => {
+                return response.id
+            });
+
+        return this.session.get(
+            'spokenAbc/skills/multiClientSkills?clientIds=' + subAccountId + '&skillType=AGENT')
+            .then((response: { data: { [x: string]: { [x: string]: any } } }) => {
+                let skillResponses = response.data['skillResponses'][subAccountId];
+                const availableSkills = [];
+                for (let skill of skillResponses) {
+                    let skillInfo = {
+                        "skillNumber": skill.number,
+                        "skillName": skill.name,
+                    };
+                    availableSkills.push(skillInfo);
+                }
+                console.log(availableSkills);
+                return availableSkills;
+            })
+    }
+
+    async sendCreateStationRequest(
+        agentStationGroupId: any,
+        subAccountId: any,
+        stationExtension: { toString: () => any },
+        agentUsername: string) {
+
+        let securityCode = this.generateSecurityCode(stationExtension);
+
+        let station = {
+            "agentStationGroupId": agentStationGroupId,
+            "clientId": subAccountId,
+            "extension": stationExtension,
+            "name": STATION_NAME,
+            "securityCode": securityCode,
+            "username": agentUsername
+        };
+
+        return this.session.post('/spokenAbc/jobs/stations', station)
+            .then((result: any) => {
+                // console.log(result.data)
+                return result
+            })
+    }
+
+    async getAgent(agentUsername: string) {
+        await this.session.login();
+
+        let agent = {};
+        try {
+            agent = await this.getAgentByUsername(agentUsername)
+        } catch (e) {
+            if (e.response.status === 404) {
+                console.log('agent ' + agentUsername + ' not found')
+            } else {
+                throw e
+            }
+        }
+
+        let subAccountId = await this.getSubAccountId();
```

Comment:
If agent not found, should execution continue here?

Reply:
we need the subaccount id for station deletion. Because station needs to be deleted before agent deletion.